internal/controller/api/v1alpha1/image_types.go

```diff
@@ -17,25 +17,28 @@ limitations under the License.
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-
 // ImageSpec defines the desired state of Image
 type ImageSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-
-	// Foo is an example field of Image. Edit Image_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// The name of the Cluster CR
+	ClusterName  corev1.ObjectReference `json:"clusterName"`
+	SSHPublicKey string                 `json:"sshPublicKey,omitempty"`
+	// The name of the secret containing the pull secret
+	PullSecretName        *corev1.SecretReference `json:"pullSecretName"`
+	IgnitionOverrides     string                  `json:"ignitionOverrides,omitempty"`
+	StaticIpConfiguration string                  `json:"staticIpConfiguration,omitempty"`
 }
 
 // ImageStatus defines the observed state of Image
 type ImageStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	State               string       `json:"state,omitempty"`
+	SizeBytes           int          `json:"sizeBytes,omitempty"`
+	DownloadUrl         int          `json:"downloadUrl,omitempty"`
+	CreationTimestamp   *metav1.Time `json:"creationTimestamp,omitempty"`
+	ExpirationTimestamp *metav1.Time `json:"expirationTimestamp,omitempty"`
```

Comment:
```suggestion
	ExpirationTime *metav1.Time `json:"expirationTimestamp,omitempty"`
```

Reply:
Consistency is boring - even though in the metadata there is creationTimestamp, the API guidelines say:
> The name of a field that specifies the time at which something occurs should be called somethingTime. Do not use stamp (e.g., creationTimestamp).

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions

Reply:
ok, I'll change it back to `ExpirationTime`

Reply:
Done